app/hooks/MenuHooks.ts

```diff
@@ -43,3 +43,30 @@ export function useMenuToggle() {
     menuRef,
   };
 }
+
+// For use in modals, dropdowns, etc.
+function useClickOutside<T extends HTMLElement>(
+  ref: React.RefObject<T>,
+  callback: () => void, // i.e. setIsActive(false)
+  isActive: boolean = true
+) {
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
+        callback();
+      }
+    };
+
+    if (isActive) {
+      document.addEventListener("mousedown", handleClickOutside);
+    }
+
+    return () => {
+      if (isActive) {
```

Comment:
Is this a redundant check here? I'm wondering because if the return is called that means the component is unmounting, therefore would likely not be _active_ anymore or present in the DOM. But I'm not sure I fully understand how the `isActive` property plays with both visibility to the user and/or presence in the DOM. It just

Reply:
I thought I put a comment here on my PR but this is here to avoid the consistent-return lint check.